esmvalcore/_recipe.py

```diff
@@ -349,26 +349,41 @@ def _add_fxvar_keys(fx_info, variable):
     return fx_variable
 
 
-def _search_fx_mip(tables, found_mip, variable, fx_info, config_user):
-    fx_files = None
-    for mip in tables:
-        fx_cmor = tables[mip].get(fx_info['short_name'])
-        if fx_cmor:
-            found_mip = True
-            fx_info['mip'] = mip
-            fx_info = _add_fxvar_keys(fx_info, variable)
-            logger.debug("For fx variable '%s', found table '%s'",
-                         fx_info['short_name'], mip)
-            fx_files = _get_input_files(fx_info, config_user)[0]
-            if fx_files:
-                logger.debug("Found fx variables '%s':\n%s",
-                             fx_info['short_name'], pformat(fx_files))
-    return found_mip, fx_info, fx_files
+def _search_fx_mip(tables, variable, fx_info, config_user):
```

Comment:
Hm. I am not sure we are on the same page here. The ambiguity does not come from the variable being defined in several tables. It comes into play when a model delivers it in several tables. For example, for CMIP6 `volcello` can be delivered in `Ofx` or (`Omon` and `Odec`). If the model delivers it only in `Ofx`, that's fine and unambiguous. If it delivers it in `Ofx` and at least one of the other two, that's an error. Delivering it for both `Omon` and `Odec` is ok; I am not sure about delivering it only for, say, `Omon`.

Reply:
Yes, I think we are and I also understand your point.

To implement it like you describe it here a whole bunch of custom rules for every fx variable for every project has to be implemented. I thought we agreed that this is not really feasible. I think simply specifying the `mip` in these cases is by far the simplest solution that works for every project.

Reply:
I don't think we would need a lot of custom rules. What I propose is basically to send the data finder out and do its worst and if it finds more than one candidate, then we barf.

Reply:
The way you described it here would mean custom rules:

> For example, for CMIP6 `volcello` can be delivered in `Ofx` or (`Omon` and `Odec`). If the model delivers it only in `Ofx`, that's fine and unambiguous. If it delivers it in `Ofx` and at least one of the other two, that's an error. Delivering it for both `Omon` and `Odec` is ok; I am not sure about delivering it only for, say, `Omon`.

How else should the tool know that `Omon` and `Odec` is ok, but `Omon` and `Ofx` is not?

Raising an error when files are found in more than one should not be an issue, I can do that. But bear in mind that this could mean that for different datasets fx variables from different tables (e.g., `IyrAnt` and `fx`) might be mixed. Is this really what we want?

Reply:
You are right, the tool would not know this. I should have phrased that better. It is true that the perfect solution, where the tool understands exactly which combinations are ok would require such rules. But my goal was to get an 80% solution by complaining only when there is actual ambiguity in the available data.

As for the second point, we absolutely want to support fx files coming from different tables for different datasets. The abovementioned `volcello` is a good example. Most models will have that in `Ofx` but [some](https://esgf-data.dkrz.de/search/cmip6-dkrz/?variable_id=volcello&table_id=Omon) have it in `Omon`.

Reply:
Alright, will implement this.